packages/react/src/organisms/sidebar/base.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from "react";
+import "../../../tailwind.css";
+import classNames from "../../utils/classNames";
+import { useThemeContext } from "../../theme/ThemeProvider";
+import resolvedStyleProps from "../../utils/resolvedStyleProps";
+import { ChevronDownIcon } from "@heroicons/react/solid";
+
+export interface SidebarProps {
+  topItem?: any;
+  navigationItems?: any;
+  onClick?: any;
+  sidebarBaseClasses?: {
+    wrapper?: string;
+    backgroundClasses?: string;
+    navItemClasses?: string;
+    navItemActiveClasses?: string;
+  };
+}
+
+const ItemRenderer = (props: any) => {
+  const [open, setOpen] = useState(false);
+  const { item, index } = props;
+
+  return item.children ? (
+    <>
+      <div
+        key={item.label}
+        onClick={(e) => {
+          e.preventDefault();
+          setOpen(!open);
+        }}
+        className={classNames(
+          open || item.current
```

Comment:
Why use both state variables and Prop as condition, when both the variables relate to the active/open state of the component?